torch/csrc/jit/codegen/cuda/graph_fuser.cpp

```diff
@@ -1276,6 +1294,53 @@ void PeepholeOptimizeShapeExpressions(Block* block) {
   }
 }
 
+// view_sizes_runtime is the profiled-ivalue argument for view-size.
+// view_sizes_constant_list is the constant list recorded during profiling runs.
+Value* guardView(
+    Node* fusion,
+    std::unordered_map<Value*, Value*>& fusion_value_to_runtime_size,
+    Node* versioning_if,
+    Node* view,
+    Value* view_sizes_runtime) {
+  // 1. Get self tensor sizes and view_sizes
+  auto self_value = view->inputs().front();
+  auto self_type = self_value->type()->cast<TensorType>();
+  auto self_sizes_constant_list = getTensorSizes(self_type);
+
+  auto view_sizes_constant_list =
+      constant_as<c10::List<int64_t>>(view->inputs().back());
+  TORCH_INTERNAL_ASSERT(view_sizes_constant_list.has_value());
+
+  // 2. Get constraints for self tensor and view_sizes
+  auto constraints = analyzeViewConstraint(
+      self_sizes_constant_list, view_sizes_constant_list->vec());
+
+  // 3. Add constraints as constant to graph
+  auto self_tensor_constraint = fusion->owningGraph()->insertConstant(
+      IValue(constraints.original_constraint));
+  self_tensor_constraint->node()->moveBefore(versioning_if);
+  auto view_sizes_constraint =
+      fusion->owningGraph()->insertConstant(IValue(constraints.new_constraint));
+  view_sizes_constraint->node()->moveBefore(versioning_if);
+
+  // 4. Create CudaFusionViewGuard using input tensor, profile_ivalue
+  // for view_sizes list, and constraints
+  TORCH_INTERNAL_ASSERT(
+      fusion_value_to_runtime_size.find(self_value) !=
+      fusion_value_to_runtime_size.end());
```

Comment:
Error message please! I promised this is the last nitpick :laughing: 
If we are not throwing an error message, the assert here is meaningless, as `unordered_map::at` throws error when the look up fails.